core/src/main/scala/cats/instances/map.scala

```diff
@@ -53,8 +57,33 @@ trait MapInstances extends cats.kernel.instances.MapInstances {
       def foldRight[A, B](fa: Map[K, A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
         Foldable.iterateRight(fa.values.iterator, lb)(f)
 
+      def tailRecM[A, B](a: A)(fn: A => Map[K, A Xor B]): Map[K, B] = {
```

Comment:
I feel like something like this might be a bit more efficient and straightforward:

```scala
  def tailRecM[A, B](a: A)(f: A => Map[K, Xor[A, B]]): Map[K, B] = {
    val bldr = Map.newBuilder[K, B]

    @tailrec def descend(k: K, m: Map[K, Xor[A, B]]): Unit =
      m.get(k) match {
        case Some(Xor.Left(a)) => descend(k, f(a))
        case Some(Xor.Right(b)) => bldr += (k, b)
        case None => ()
      }

    m.foreach { case (k, a) => descend(k, a) }
    bldr.result
  }
```